src/cpp/include/openvino/genai/generation_handle.hpp

```diff
@@ -32,6 +32,12 @@ struct EncodedGenerationResult {
     GenerationStatus m_status = GenerationStatus::RUNNING;
 };
 
+enum class GenerationFinishReason {
```

Comment:
`GenerationStatus` is a status for the whole request while `GenerationFinishReason` is per sequence, so for example when we have beam search or n > 1 we can read generation status for entire request (that would be finished) and get finish reason for every output separately (in case some of the beams hit max_new_tokens and some of them stopped naturally due to EOS token for example). 

I also considered using `SequenceStatus`, but this class is internal and we need a one that's available for the user.